ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-commands.sh

```diff
@@ -6,8 +6,166 @@ set -o pipefail
+  wait_for_stack "${STACK_NAME_CAGW}"
+
+  echo "${STACK_NAME_CAGW}" >> "${SHARED_DIR}/sharednetwork_stackname_edge_cagw"
+}
+
+function create_stack_edge_subnets() {
+
+  template_file="01.99_subnet.yaml"
+  template_path="/tmp/${template_file}"
+  template_url="${TEMPLATE_BASE_URL}"/"${template_file}"
+
+  show "Downloading CloudFormation template to create Public and Private subnets from ${template_url}"
+  curl -sL "${template_url}" -o $template_path
+
+  edge_zone_name=$(< "${SHARED_DIR}"/edge-zone-name.txt)
+  show "Edge Zone selected: ${edge_zone_name}"
+
+  vpc_rtb_pub=$(aws --region "$REGION" cloudformation describe-stacks \
+    --stack-name "${STACK_NAME_CAGW}" \
+    | jq -r '.Stacks[0].Outputs[] | select(.OutputKey=="PublicRouteTableId").OutputValue' )
+
+  #> Select the first route table from the list
+  vpc_rtb_priv=$(aws --region "$REGION" cloudformation describe-stacks \
+    --stack-name "${STACK_NAME_VPC}" \
+    | jq -r '.Stacks[0].Outputs[]
+      | select(.OutputKey=="PrivateRouteTableIds").OutputValue
+      | split(",")[0] | split("=")[1]' \
+  )
+
+  subnet_cidr_pub="10.0.128.0/24"
+  subnet_cidr_priv="10.0.129.0/24"
+
+  cat <<EOF
+REGION=$REGION
+VPC_ID=$VPC_ID
+edge_zone_name=$edge_zone_name
+vpc_rtb_pub=$vpc_rtb_pub
+vpc_rtb_priv=$vpc_rtb_priv
+subnet_cidr_pub=$subnet_cidr_pub
+subnet_cidr_priv=$subnet_cidr_priv
+EOF
+
+  STACK_NAME_SUBNETS=${CLUSTER_NAME}-subnets-${edge_zone_name/${REGION}-/}
+  show "Creating stack: ${STACK_NAME_SUBNETS}"
+  aws cloudformation create-stack \
+    --region "${REGION}" \
+    --stack-name "${STACK_NAME_SUBNETS}" \
+    --template-body file://$template_path \
+    --parameters \
+      ParameterKey=VpcId,ParameterValue="${VPC_ID}" \
+      ParameterKey=ClusterName,ParameterValue="${CLUSTER_NAME}" \
+      ParameterKey=ZoneName,ParameterValue="${edge_zone_name}" \
+      ParameterKey=PublicRouteTableId,ParameterValue="${vpc_rtb_pub}" \
+      ParameterKey=PublicSubnetCidr,ParameterValue="${subnet_cidr_pub}" \
+      ParameterKey=PrivateRouteTableId,ParameterValue="${vpc_rtb_priv}" \
+      ParameterKey=PrivateSubnetCidr,ParameterValue="${subnet_cidr_priv}"
+
+  show "Created stack: ${STACK_NAME_SUBNETS}"
+  wait_for_stack "${STACK_NAME_SUBNETS}"
+
+  subnet_edge=$(aws --region "${REGION}" cloudformation describe-stacks --stack-name "${STACK_NAME_SUBNETS}" --query "Stacks[0].Outputs[?OutputKey=='PrivateSubnetId'].OutputValue" --output text | tr ',' '\n')
+  subnets_arr=$(jq -c ". + [\"$subnet_edge\"]" <(echo "$subnets_arr"))
+  show "Subnets (including edge zones): ${subnets_arr}"
+
+  echo "${STACK_NAME_SUBNETS}" >> "${SHARED_DIR}/sharednetwork_stackname_edge_subnets"
+}
+
+if [[ "${EDGE_ZONE_TYPE-}" == "wavelength-zone" ]]; then
+  # TODO(mtulio/mrbraga): should use installer@main once the PR is merged:
+  # https://github.com/openshift/installer/pull/7652
+  TEMPLATE_BASE_URL="https://raw.githubusercontent.com/mtulio/installer/edge-aws-wavelength-zone-byovpc/upi/aws/cloudformation"
```

Comment:
Why the fork? Can we include this in installer repo?

Reply:
@vrutkovs this is a temporary URL while the PR isn't merged: https://github.com/openshift/installer/pull/7652
Ideally, it could use the local path of the cloned repository, but I am not sure if it is possible in CI. Thoughts?

Reply:
>Ideally, it could use the local path of the cloned repository, but I am not sure if it is possible in CI. Thoughts?

CI just runs images, so if you include the template in https://github.com/openshift/installer/blob/master/images/installer/Dockerfile.upi.ci, it can be used here.

I don't think I can lgtm this PR until it uses master branch / local source

Reply:
@vrutkovs , that's a great idea and it will improve a lot the existing flow as we'll be able to test the template in installer PRs. I will submit an isolated PR to move the existing codebase to consume templates in the image.

Once it is merged, I will return to this PR.

Reply:
PR opened to validate the usage in local build: https://github.com/openshift/release/pull/45481

Reply:
PR https://github.com/openshift/release/pull/45481 has been merged. Curl removed and now we are using templates from the CI images.